app/models/waiting_list.rb

```diff
@@ -3,4 +3,13 @@ class WaitingList < ApplicationRecord
   belongs_to :course_session, optional: true
 
   scope :current, -> { where(course_session: nil).includes(:user) }
+
+  class << self
+    def notify_course_session_open(course_session)
```

Comment:
yup thats great! reason why i wanted that, in the future we might have timeouts here , as we don't know how long sending email for each shopper takes via "single email" api that is used right now, so we likely will move it to "job" etc.